Documentation/README_ObjectManipulator.md

```diff
@@ -0,0 +1,242 @@
+
+### Physics
+
+#### Release behavior
+
+Specify which physical properties a manipulated object should keep upon release. Requires a rigidbody component to be on that object. Because this property is a flag, you can select any number of options.
+
+* *Keep Velocity*: When the object is released, if this option is selected it will keep its linear velocity.
+* *Keep Angular Velocity*: When the object is released, if this option is selected it will keep its angular velocity.
+
+### Smoothing
+
+#### Smoothing active
+Specifies whether smoothing is active.
+
+#### Move lerp time
+Amount of smoothing to apply to the movement. Smoothing of 0 means no smoothing. Max value means no change to value.
+
+#### Rotate lerp time
+Amount of smoothing to apply to the rotation. Smoothing of 0 means no smoothing. Max value means no change to value.
+
+#### Scale lerp time
+Amount of smoothing to apply to the scale. Smoothing of 0 means no smoothing. Max value means no change to value.
+
+### Manipulation events
+
+Manipulation handler provides the following events:
+
+* *OnManipulationStarted*: Fired when manipulation starts.
+* *OnManipulationEnded*: Fires when the manipulation ends.
+* *OnHoverStarted*: Fires when a hand / controller hovers the manipulatable, near or far.
+* *OnHoverEnded*: Fires when a hand / controller un-hovers the manipulatable, near or far.
+
+## Transform constraints
+
+Constraints can be used to limit manipulation in some way. For example, you may wish to allow rotation, but you want to keep the object upright. In this case you can attach a `RotationAxisConstraint` and limit rotation to y-axis rotation. MRTK provides a number of constraints, all of which are described below.
+
+It is also possible to create your own constraints and use them to create unique manipulation behaviour that you may need for your application. To do this, create a script that inherits from [`TransformConstraint`](../Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/Constraints/TransformConstraint.cs) and implement the abstract `ConstraintType` property and the abstract `ApplyConstraint` method. Upon adding your new constraint to the object, it should constrain manipulation in the way you defined. Your new constraint should also show in the object manipulator [constraint fields](#constraints).
+
+All of the constraints provided by MRTK share the following properties:
+
+#### Target Transform
+
+Transform of the manipulated object being constrained. This should be the same as the ObjectManipulator [*Host transform*](#host-transform). Defaults to the object of the component.
+
+#### Hand Type
+
+Specifies whether the constraint is used for one handed, two handed or both kinds of manipulation. Because this property is a flag, you can select both options.
+
+* *One handed*: Constraint will be used during one handed manipulation if selected.
```

Comment:
Given that the manipulation type earlier is a filter on what will happen, I assume that if you specify manipulation type == one handed, but then add a constraint that is two handed, it is basically going to be a no-op right?

Reply:
Yeah, at the moment there's no warning or anything for that. We can create an issue if you think it's worthwhile.